Add Android blog post answering common questions and sharing more links and info

finagolfin opened the pull request:
The Android workgroup has been monitoring aggregators and other forums and seeing more detailed questions about the new SDK snapshots. Marc and I put together this post with more info, [with input from others in the workgroup and long-time Swift on Android users](https://github.com/swift-android-sdk/swift-org-website/pull/1), and the SWWG asked us to put it up for public review now.

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+swift-java project's `jextract` tool and its [new support for generating such
+JNI bindings for you](/blog/gsoc-2025-showcase-swift-java/) comes in. Please
+watch its author Mads Odgaard's [Server-Side Swift Conference talk from last month](https://www.youtube.com/watch?v=tOH6V1IvTAc)
+and try out the tool for yourself with [this example Android app that he put together](https://github.com/swiftlang/swift-android-examples/tree/main/hello-swift-java).
```
```suggestion
and try out the tool for yourself with [this example Android app he put together](https://github.com/swiftlang/swift-android-examples/tree/main/hello-swift-java).
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+Since the announcement of [the preview Swift SDK for Android last month](/blog/nightly-swift-sdk-for-android/),
+the Android workgroup has seen a lot of questions about how it works and what's next. Please read on for those answers.
+
+### Swift in the World of Android
```
```suggestion
## How Swift works on Android
```

davelester replied:
Circling back, this and the additional heading revision are editorially consistent with the style of the blog. The swift.org blog uses headings as signposts that are descriptive in nature and make posts easy to scan.

Prose within sections can be adjusted to have more energy and personality, keeping in mind this particular post is on behalf of a workgroup vs an individual. If you can accept these changes, happy to review revised copy in the body to address your concern.

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+Android apps daily shows both the stability of Swift on Android and the
+practical benefits it provides to development teams sharing code across platforms.
+
+### Coming up Next
```
```suggestion
## What's next for Swift on Android
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+watch its author Mads Odgaard's [Server-Side Swift Conference talk from last month](https://www.youtube.com/watch?v=tOH6V1IvTAc)
+and try out the tool for yourself with [this example Android app that he put together](https://github.com/swiftlang/swift-android-examples/tree/main/hello-swift-java).
+
+### Swift in Android app stores
```
```suggestion
## Swift on Android in production
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+the viability of this approach:
+
+- [Spark](https://play.google.com/store/apps/details?id=com.readdle.spark) - A popular email client using Swift to share code between mobile iOS/Android and desktop macOS/Windows versions
+- [flowkey](https://play.google.com/store/apps/details?id=com.flowkey.app) - This interactive piano learning app has been built with Swift on Android since the open-source release
```
```suggestion
- [flowkey](https://play.google.com/store/apps/details?id=com.flowkey.app) - An interactive piano learning app, built with Swift on Android
```

davelester replied:
Updated copy to be more consistent with other list items, and removed "since the open-source release" since it was unclear what release this was referring to. If it was kept, we should clarify (which may be too lengthy?) and remove the hyphen from open source.

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+### Coming up Next
+
+Swift on Android [first got started as soon as the language was open-sourced a decade ago](https://lists.swift.org/pipermail/swift-dev/Week-of-Mon-20151207/000171.html),
+but it is by no means done. [The Android project board lists areas we are working on](https://github.com/orgs/swiftlang/projects/17)
```
```suggestion
and continue to this day. [The Android project board lists areas the Android Workgroup is working on](https://github.com/orgs/swiftlang/projects/17)
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+- [MediQuo](https://play.google.com/store/apps/details?id=com.mediquo.main) - A healthcare app leveraging Swift for cross-platform development
+- [Naturitas](https://play.google.com/store/apps/details?id=com.naturitas.android) - An organic products marketplace running Swift in production
+
+These aren't experimental apps or proofs of concept—they're real businesses serving
```
A more direct way of saying this is just to assert that these are real businesses, vs starting off by stating what they are not.

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+
+### Swift in Android app stores
+
+While Swift on Android may seem new, it's important to recognize that there are
```
```suggestion
While Swift on Android may seem new, it's important to recognize that Android
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+### Swift in Android app stores
+
+While Swift on Android may seem new, it's important to recognize that there are
+already millions of Android devices running Swift today. Production apps with
```
```suggestion
apps running Swift has been downloaded millions of time, and used for years.
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+
+While Swift on Android may seem new, it's important to recognize that there are
+already millions of Android devices running Swift today. Production apps with
+substantial user bases have been shipping Swift on Android for years, proving
```
```suggestion
Here are a few examples of Android apps using Swift:
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+While Swift on Android may seem new, it's important to recognize that there are
+already millions of Android devices running Swift today. Production apps with
+substantial user bases have been shipping Swift on Android for years, proving
+the viability of this approach:
```
```suggestion
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+curate a list of cross-platform UI tools. See [our recent post in the Swift forums](https://forums.swift.org/t/swift-gui-toolkits-for-android/83337)
+listing a handful of popular and in-progress options.
+
+### Diving in
```
```suggestion
## Getting involved
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+### Swift in the World of Android
+
+Swift is a natively-compiled language to machine code and Android is no different.
+That renders it on par with C and C++ code built using the Android NDK, which are
```
```suggestion
platforms. This gives Swift similar performance to C and C++ code using the Android NDK, 
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+languages more geared towards performance, while Swift essays a happier balance between
+performance, safety, and usability. To enable that, Swift apps must bundle a native
+runtime for Android that implements many of its features, including its standard library and
+core libraries like Dispatch and [Foundation](/blog/foundation-preview-now-available/).
```
```suggestion
and [Foundation](/blog/foundation-preview-now-available/).
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+and was [interviewed by Swift Toolkit last month](https://www.swifttoolkit.dev/posts/dc-pier).
+
+Swift on Android has been a community effort for the last decade, snowballing
+from that initial patch to many businesses and developers making their livelihood
```
```suggestion
from an initial patch to a community of contributors and the Swift project's [Android Workgroup](https://www.swift.org/android-workgroup/).
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+
+Swift on Android has been a community effort for the last decade, snowballing
+from that initial patch to many businesses and developers making their livelihood
+with it today. [Try out the preview SDK snapshots](/documentation/articles/swift-sdk-for-android-getting-started.html)
```
```suggestion
[Try out the preview SDK snapshots](/documentation/articles/swift-sdk-for-android-getting-started.html)
```

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+into editors like [Visual Studio Code](/blog/gsoc-2025-showcase-swiftly-support-in-vscode/),
+Android Studio, and [CodeEdit](https://www.codeedit.app/), another issue on our board.
+
+### Sharing Logic Versus Sharing UI
```
Do we have a few concise examples of the type of business logic that you'd commonly see shared between platforms? It'd be great if we could outline a few examples in one paragraph, before mentioning cross-platform GUIs in a second paragraph.

finagolfin replied:
Again disagree with the heading change, but good point on the business logic example, will add that.

finagolfin replied:
Added a clause on the value of sharing code across platforms but kept the heading, which describes this section well. Added some code in a different section, though may be too much code now. :wink:

finagolfin replied:
Removed the mention of working with GUI devs in the latest commit

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+runtime for Android that implements many of its features, including its standard library and
+core libraries like Dispatch and [Foundation](/blog/foundation-preview-now-available/).
+
+However, since most Android APIs are only made available through Java and Kotlin
```
LGTM @ktoso!

finagolfin commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+gave [a great talk at NSSpain 2025 a couple months ago](https://youtu.be/EIGl6GOo210),
+and was [interviewed by Swift Toolkit last month](https://www.swifttoolkit.dev/posts/dc-pier).
+
+Swift on Android has been a community effort for the last decade, snowballing
```
Prefer the real-world term

davelester commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,77 @@
+category: "Developer Tools"
+---
+
+Since the announcement of [the preview Swift SDK for Android last month](/blog/nightly-swift-sdk-for-android/),
```
```suggestion
Since the announcement of [the preview Swift SDK for Android](/blog/nightly-swift-sdk-for-android/),
```

finagolfin commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+
+### Diving in
+
+Finally, we intend to bring you interviews and information from those using Swift
```
Got rid of this

finagolfin commented on _posts/2025-11-21-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,79 @@
+### Diving in
+
+Finally, we intend to bring you interviews and information from those using Swift
+on Android already, as pioneering companies like [Readdle](https://readdle.com/blog/swift-for-android-our-experience-and-tools)
```
Changed this to path-breaker instead

finagolfin replied:
Changed this to "founding contributors" in the latest commit

finagolfin commented:
Updated this with most everything discussed, including a new code sample too. Three editor's notes remain for external links that are still being worked on, that we hope to get in before this post is published, but this post itself is ready for final review. I have changed the tentative publishing date to Thursday morning PST, which should give us enough time to finish up review and those last few tasks, or push it to Friday if needed.

@compnerd, let me know if I should link the 6.3 Windows toolchain snapshots too, ie if they include the Android SDKs also.

ktoso commented on _posts/2025-12-18-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,157 @@
+
+A [community member recently contributed an example app](https://github.com/swiftlang/swift-android-examples/pull/24)
+that builds C++ using CMake and links it with Swift, using Swift's automated JNI bridging
+to Java instead. (editor: not sure we will get this in, but might be a good example to highlight for some)
```
~~There's also the https://github.com/swiftlang/swift-android-examples/pull/25 which showcases bi-directional uses with swift-java, perhaps this section should become a "check our latest examples"?~~ 

Oh, I see it is linked above, that's ok

ktoso commented on _posts/2025-12-18-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,157 @@
+
+The [`jextract` tool gained a JNI mode this summer](/blog/gsoc-2025-showcase-swift-java/):
+you can now watch its author Mads Odgaard's [Server Side Swift Conference talk from a couple months ago](https://www.youtube.com/watch?v=tOH6V1IvTAc)
+and try out [his new weather example app in the Android examples repository](
```
```suggestion
and try out [the new weather example app in the Android examples repository](
```

It's true that it's "his" example, but more than that, it is a Swift project example, so I'd phrase it as "the example"

finagolfin replied:
He wrote it, so giving him credit, but don't really care 🤷‍♂️

finagolfin replied:
Changed this to "the new weather example app he submitted in," which gives him credit and indicates to readers that it is worth perusing, since it came from the main JNI guy.

ktoso commented on _posts/2025-12-18-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,157 @@
+@available(Android 27, *)
+func hello27() {
+    print("Hello, world!")
+}
```
It's probably enough to just showcase one method to get the point across

finagolfin commented on _posts/2025-12-18-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,157 @@
+
+[An Android workflow](https://github.com/swiftlang/github-workflows/pull/172) was
+added to the official Swift workflows for GitHub months ago, allowing you to easily
+try building your Swift packages with the Swift SDK for Android, and work is underway to let you
```
It might, see #1198

finagolfin commented:
> Right now there are a number markdown links where the text that's highlighted is quite long; I can open up specific suggestions to change this (easy merges, I hope) but that'd bring the style more in line with other posts.

No problem with shortening some, so go ahead and list the ones you want, but prefer not to make them too short, as longer links are more clickable on mobile browsers, which is where the vast majority of traffic is these days. I know I still made a couple too short, but only because I didn't see a way to make those larger.

> This is currently under the "Ongoing work" section but easily overlooked -- pulling it up to the intro would make it much more visible.

That was intentional: I want serious, engaged contributors who actually read the middle of the post. Unfortunately, a lot of OSS projects get people who say they want to pitch in, then disappear and never do anything: we've gotten a small taste of that with our recent publicity too.

> Announce 6.3 SDK snapshots of the Swift SDK for Android. This is currently added in a section at the bottom of the post, but bringing it right to the top would be fantastic.

I was going for a "one more thing" Apple keynote effect, :wink: but happy to mention it twice if you think that's better.

finagolfin commented on _posts/2025-12-18-exploring-the-swift-sdk-for-android.md:
````diff
@@ -0,0 +1,157 @@
+    }
+}
+```
+(editor: should we slim this down?)
````
Slimmed it down to only check one API level at runtime

finagolfin commented:
I added most of the changes asked for, and teased the new 6.3 SDK snapshots at the top.

davelester commented on _posts/2025-12-18-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,141 @@
+---
+
+Since the announcement of [the preview Swift SDK for Android](/blog/nightly-swift-sdk-for-android/),
+the Android workgroup has seen a lot of questions about how it works and what's next.
```
```suggestion
the Android workgroup has seen a lot of interest in how it works and what's next.
```

davelester replied:
Not 100% necessary, but the following sentence says "questions" so this adds a bit of variety to the prose and also gives a positive framing "interest" versus questions which may have negative connotations.

davelester commented on _posts/2025-12-18-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,141 @@
+allows Swift to seamlessly integrate with the Android platform.
+
+The [`jextract` tool gained a JNI mode this summer](/blog/gsoc-2025-showcase-swift-java/):
+you can watch its author Mads Odgaard's [Server Side Swift Conference talk from a couple months ago](https://www.youtube.com/watch?v=tOH6V1IvTAc)
```
```suggestion
you can watch its author Mads Odgaard's [Server-Side Swift Conference talk](https://www.youtube.com/watch?v=tOH6V1IvTAc)
```

davelester commented on _posts/2025-12-18-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,141 @@
+
+The [`jextract` tool gained a JNI mode this summer](/blog/gsoc-2025-showcase-swift-java/):
+you can watch its author Mads Odgaard's [Server Side Swift Conference talk from a couple months ago](https://www.youtube.com/watch?v=tOH6V1IvTAc)
+and try out [the new weather app he submitted in the Android examples repository](
```
```suggestion
and try out the [example weather app](https://github.com/swiftlang/swift-android-examples/tree/main/swift-java-weather-app) he submitted in the Android examples repository].
```

davelester commented on _posts/2025-12-18-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,141 @@
+
+Swift allows you to target many platforms with the same business logic, and Swift
+on Android expands that much more, but we do not provide a cross-platform GUI toolkit.
+As we write in [our draft vision document](https://github.com/swiftlang/swift-evolution/blob/807b844be42db582e434d1667fc907ae7a7a8775/visions/android.md),
```
```suggestion
As we shared in [our draft vision document](https://github.com/swiftlang/swift-evolution/blob/807b844be42db582e434d1667fc907ae7a7a8775/visions/android.md),
```

davelester commented on _posts/2025-12-18-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,141 @@
+## Sharing Logic Versus Sharing UI
+
+Swift allows you to target many platforms with the same business logic, and Swift
+on Android expands that much more, but we do not provide a cross-platform GUI toolkit.
```
right now the mention of not providing a cross-platform GUI toolkit is spread across two sentences. going to suggest an edit to combine these statement so it's simpler and clearer.

davelester commented on _posts/2025-12-18-exploring-the-swift-sdk-for-android.md:
```diff
@@ -0,0 +1,141 @@
+the Android workgroup has no plans to create such a GUI toolkit, but will instead
+curate a list of cross-platform UI tools from the community.
+
+See [our recent post in the Swift forums](https://forums.swift.org/t/swift-gui-toolkits-for-android/83337)
```
```suggestion
Our [recent post in the Swift forums](https://forums.swift.org/t/swift-gui-toolkits-for-android/83337)
```
